src/FxCop/System.Globalization.Analyzers/Test/Globalization/CA1305SpecifyIFormatProviderTests.cs

```diff
@@ -173,53 +308,201 @@ void M(string strA)
+                                                 "Public Sub M(strA As String)",
+                                                 "Public Shared Overloads Function Parse(s As String, style As System.Globalization.NumberStyles, provider As System.IFormatProvider) As Long"));
+        }
+
+        [Fact]
+        public void CA1305StringFormat()
+        {
+            VerifyCSharp(@"using System;
+using System.Globalization;
+using System.Threading;
+
+sealed class C
+{
+    void M(string strA, string strB)
+    {
         string str1 = string.Format(Thread.CurrentThread.CurrentUICulture, ""Foo {0}"", strA);
         string str2 = string.Format(CultureInfo.InstalledUICulture, ""Foo {0} {1}"", strA, strB);
+    }
+ 
+}",
+            GetCA1305CSharpDefaultResultAt(9, 23, MessageUICultureString,
+                                                  "C.M(string, string)",
+                                                  "Thread.CurrentThread.CurrentUICulture",
+                                                  "string.Format(System.IFormatProvider, string, object)"),
+            GetCA1305CSharpDefaultResultAt(10, 23, MessageUICultureString,
+                                                  "C.M(string, string)",
+                                                  "CultureInfo.InstalledUICulture",
+                                                  "string.Format(System.IFormatProvider, string, object, object)"));
+
+            VerifyBasic(@"
+Imports System
+Imports System.Globalization
+Imports System.Threading
+
+
+Public Module C
+    Public Sub M(strA As String, strB As String)
+        Dim str1 As String = string.Format(Thread.CurrentThread.CurrentUICulture, ""Foo {0}"", strA)
+        Dim str2 As String = string.Format(CultureInfo.InstalledUICulture, ""Foo {0} {1}"", strA, strB)
+    End Sub
+End Module
+",
+            GetCA1305BasicDefaultResultAt(9, 30, MessageUICultureString,
+                                                 "Public Sub M(strA As String, strB As String)",
+                                                 "Thread.CurrentThread.CurrentUICulture",
+                                                 "Public Shared Overloads Function Format(provider As System.IFormatProvider, format As String, arg0 As Object) As String"),
+            GetCA1305BasicDefaultResultAt(10, 30, MessageUICultureString,
+                                                  "Public Sub M(strA As String, strB As String)",
+                                                  "CultureInfo.InstalledUICulture",
+                                                  "Public Shared Overloads Function Format(provider As System.IFormatProvider, format As String, arg0 As Object, arg1 As Object) As String"));
+        }
+
+        [Fact]
+        public void CA1305IFormatProvider_LeadingCultureInfo()
+        {
+            VerifyCSharp(@"using System;
```

Comment:
Start to see some inconsistent style here. Shall we put the first statement on the next line of '@"' or same line? Once we agrees, we can follow this convention.